### ISSUE TITLE
usePagination implementation

### ISSUE THREAD

**Issue by deebov** [adobe/react-spectrum#1748]
# 🙋 Feature Request

I would like to build `usePagination` that handles the state and behavior for a pagination component. I'd like to know if react-aria team has already started working on this feature. If not, how could I start working on this feature? I think starting with the a11y design (such as which aria attributes to use, how components should behave) would be a good idea. Also I think we could reuse some of the statetely hooks.

**Comment by LFDanLu** [comment]
Hey, thanks for the offer to help! We haven't done any finalized work yet for `usePagination`, the hook we have in our code base right now is a bit outdated now. Feel free to look into the a11y design, would love to see what you come up with regarding the api, the accessibility, and the functionality that `usePagination` would have. You could post anything you want reviewed in this issue or in a RFC/pull and tag one of the team members for visiblity.

**Comment by LFDanLu** [comment]
I'm curious though, what stately hooks did you have in mind to reuse here? Do you have any initial thoughts or rough ideas on what functionality the usePagination hook would provide and what the api might look like?

I found https://www.w3.org/TR/wai-aria-practices-1.2/#aria_lh_navigation for some navigational guidelines but it doesn't really mention what the aria expectations would be for the nav buttons themselves (e.g. Next/Prev button would need aria-labels). @majornista any thoughts on the above? Maybe that is the wrong spec to look at?

**Comment by dannify** [comment]
Closing as inactive. This might be picked up by the team at a later date
